dietpi/dietpi-software

```diff
@@ -13330,6 +13471,21 @@ _EOF_
 			rm /etc/systemd/system/spotify-connect-web.service
 
 		fi
+    
+    	software_id=182 # Unbound
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == -1 )); then
+
+			Banner_Uninstalling
+
+			G_AGP unbound
+      
+			if grep -q 'PIHOLE_DNS_1=127.0.0.1#5353' /etc/pihole/setupVars.conf
+			then
+				G_DIETPI-NOTIFY 2 'Setting Pi-hole upstream DNS server (previously Unbound) to chosen DNS server in dietpi-config'
+				G_CONFIG_INJECT 'PIHOLE_DNS_1=' "PIHOLE_DNS_1=$(mawk '/^[[:blank:]]*nameserver[[:blank:]]/{print $2;exit}' /etc/resolv.conf)" /etc/pihole/setupVars.conf
```

Comment:
It is not recommended, but it might be possible that Pi-hole is the DNS server and adding Pi-hole as it's own upstream leads to a DNS loop.

Not 100% sure, but actually I think we can assume that admins remember to change the upstream DNS in Pi-hole when uninstalling unbound, what you think? Otherwise we apply the DietPi default `9.9.9.9`?

Of course we could also check if it's 127.0.0.1 or practically any address starting with `127.` which are all handled the same way as loopback addresses. But maybe this is too much effort for something too trivial? At least the script cannot fail at this stage, even if Pi-hole would be the DietPi's own DNS resolver and unbound was it's own configured upstream, since no more requests are done from here on.

Reply:
I'm just going to replace it with a G_WHIP_MSG as a reminder.